libmcount/mcount.c

```diff
@@ -787,15 +787,21 @@ static void mcount_finish(void)
 
 static bool mcount_check_rstack(struct mcount_thread_data *mtdp)
 {
-	if (mtdp->idx >= mcount_rstack_max) {
-		static bool warned = false;
+	static bool warned = false;
 
+	if (mtdp->idx >= mcount_rstack_max) {
 		if (!warned) {
+			struct mcount_ret_stack *rstack;
+
 			pr_warn("too deeply nested calls: %d\n", mtdp->idx);
+			/* flush current rstack */
+			rstack = &mtdp->rstack[mtdp->idx - 1];
+			record_trace_data(mtdp, rstack, NULL);
```

Comment:
And I think we need to dump rstack for each task separately.  Why not having a warned variable in the mtd?

Reply:
I just moved `warned` to `mcount_thread_data`. Thanks.